clinicadl/clinicadl/tools/tsv/demographics_analysis.py

```diff
@@ -9,51 +9,42 @@
 __email__ = "elina.ts@free.fr"
 __status__ = "Completed"
 
-import argparse
 import pandas as pd
 from .tsv_utils import first_session, next_session, add_demographics
 import os
 from os import path
 import numpy as np
 
-if __name__ == "__main__":
 
-    parser = argparse.ArgumentParser(description="Argparser for data formatting")
+def demographics_analysis(merged_tsv, formatted_data_path, results_path,
+                          diagnoses, mmse_name="MMS", age_name="age"):
 
-    # Mandatory arguments
-    parser.add_argument("merged_tsv", type=str,
-                        help="Path to the file obtained by the command clinica iotools merge-tsv.")
-    parser.add_argument("formatted_data_path", type=str,
-                        help="Path to the folder containing formatted data.")
-    parser.add_argument("results_path", type=str,
-                        help="Path to the resulting tsv file (filename included).")
-
-    # Modality selection
-    parser.add_argument("--diagnoses", nargs="+", type=str, choices=['AD', 'CN', 'MCI', 'sMCI', 'pMCI'],
-                        default=['AD', 'CN'], help="Diagnosis that must be selected from the tsv file")
-    parser.add_argument("--mmse_name", type=str, default="MMS",
-                        help="Name of the variable related to the MMSE score in the merged tsv file.")
-
-    args = parser.parse_args()
-
-    merged_df = pd.read_csv(args.merged_tsv, sep='\t')
+    merged_df = pd.read_csv(merged_tsv, sep='\t')
     merged_df.set_index(['participant_id', 'session_id'], inplace=True)
-    parent_directory = path.abspath(path.join(args.results_path, os.pardir))
+    parent_directory = path.abspath(path.join(results_path, os.pardir))
     if not path.exists(parent_directory):
         os.makedirs(parent_directory)
 
-    fields_dict = {'age': 'age_bl', 'sex': 'sex', 'MMSE': args.mmse_name, 'CDR': 'cdr_global'}
+    fields_dict = {'age': age_name, 'sex': 'sex', 'MMSE': mmse_name, 'CDR': 'cdr_global'}
+
+    if age_name not in merged_tsv.columns.values:
+        raise ValueError("The column corresponding to age is not labelled as %s. "
+                         "Please change the parameters." % age_name)
+
+    if mmse_name not in merged_tsv.columns.values:
+        raise ValueError("The column corresponding to mmse is not labelled as %s. "
+                         "Please change the parameters." % mmse_name)
 
     columns = ['n_subjects', 'mean_age', 'std_age', 'min_age', 'max_age', 'sexF', 'sexM', 'mean_MMSE', 'std_MMSE',
```

Comment:
Shorter lines improves readability of code, e.g.:
```suggestion
    columns = ['n_subjects', 'mean_age', 'std_age',
               'min_age', 'max_age', 'sexF',
               'sexM', 'mean_MMSE', 'std_MMSE',
               'min_MMSE', 'max_MMSE', 'CDR_0',
               'CDR_0.5', 'CDR_1', 'CDR_2',
               'CDR_3', 'mean_scans', 'std_scans',
               'n_scans']
```